data-prepper-plugins/mongodb/src/main/java/org/opensearch/dataprepper/plugins/mongo/stream/StreamWorker.java

```diff
@@ -175,35 +176,55 @@ public void processStream(final StreamPartition streamPartition) {
+                            if (isCRUDOperation(operationType)) {
+                                final String record;
+                                if (OperationType.DELETE == operationType) {
+                                    record = document.getDocumentKey().toJson(writerSettings);
+                                } else {
+                                    record = document.getFullDocument().toJson(writerSettings);
+                                }
+                                final long eventCreationTimeMillis = document.getClusterTime().getTime() * 1000L;
+                                final long bytes = record.getBytes().length;
+                                bytesReceivedSummary.record(bytes);
 
-                        records.add(event);
-                        recordCount += 1;
+                                checkPointToken = document.getResumeToken().toJson(writerSettings);
+                                // TODO fix eventVersionNumber
+                                final Event event = recordConverter.convert(record, eventCreationTimeMillis, eventCreationTimeMillis, document.getOperationTypeString());
+                                records.add(event);
+                                recordCount += 1;
 
-                        if ((recordCount % recordFlushBatchSize == 0) || (System.currentTimeMillis() - lastBufferWriteTime >= bufferWriteIntervalInMs)) {
-                            LOG.debug("Write to buffer for line {} to {}", (recordCount - recordFlushBatchSize), recordCount);
-                            writeToBuffer(records, checkPointToken, recordCount);
-                            lastLocalCheckpoint = checkPointToken;
-                            lastLocalRecordCount = recordCount;
-                            lastBufferWriteTime = System.currentTimeMillis();
-                            records.clear();
+                                if ((recordCount % recordFlushBatchSize == 0) || (System.currentTimeMillis() - lastBufferWriteTime >= bufferWriteIntervalInMs)) {
+                                    LOG.debug("Write to buffer for line {} to {}", (recordCount - recordFlushBatchSize), recordCount);
+                                    writeToBuffer(records, checkPointToken, recordCount);
+                                    lastLocalCheckpoint = checkPointToken;
+                                    lastLocalRecordCount = recordCount;
+                                    lastBufferWriteTime = System.currentTimeMillis();
+                                    records.clear();
+                                }
+                            } else if(shouldTerminateChangeStream(operationType)){
+                                stop();
+                                partitionCheckpoint.resetCheckpoint();
+                                LOG.warn("The change stream is invalid due to stream operation type {}. Stopping the change stream.", operationType);
```

Comment:
Will this message be misleading? Will users know that we will continue reading from the change stream?

Reply:
Update the log message